src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -303,18 +303,15 @@ const GraphComponent *GafferScene::EditScopeAlgo::transformEditReadOnlyReason( c
 		return MetadataAlgo::readOnlyReason( rows );
 	}
 
-	// As any of our columns could be locked, pick the first one.
-	for( const auto child : { g_translate, g_rotate, g_scale, g_pivot } )
+	for( const auto &plug : Plug::RecursiveRange( *row->cellsPlug() ) )
 	{
-		const Spreadsheet::CellPlug *cell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( child );
-		if( !cell ) { continue; }
-
-		if( const auto reason = MetadataAlgo::readOnlyReason( cell->valuePlug<V3fPlug>() ) )
+		if( MetadataAlgo::getReadOnly( plug.get() ) )
 		{
-			return reason;
+			return MetadataAlgo::readOnlyReason( plug.get() );
 		}
 	}
-	return nullptr;
+
+	return MetadataAlgo::readOnlyReason( row->cellsPlug() );
```

Comment:
I think this is the check that we can move in front of the recursion.

Reply:
See [update](https://github.com/GafferHQ/gaffer/pull/3840/commits/c669a77363b8fd831db6ddd4a18866d533852c0c)